framework/src/node/state_machine/types.ts

```diff
@@ -26,24 +27,14 @@ export interface ImmutableSubStore {
 	get(key: Buffer): Promise<Buffer>;
 	getWithSchema<T>(key: Buffer, schema: Schema): Promise<T>;
 	has(key: Buffer): Promise<boolean>;
-	iterate(input: {
-		start: Buffer;
-		end: Buffer;
-		limit?: number;
-		reverse?: boolean;
-	}): Promise<Buffer[]>;
-	iterateWithSchema<T>(input: {
-		start: Buffer;
-		end: Buffer;
-		limit?: number;
-		reverse?: boolean;
-	}): Promise<T[]>;
+	iterate(input: IterateOptions): Promise<{ key: Buffer; value: Buffer }[]>;
+	iterateWithSchema<T>(input: IterateOptions, schema: Schema): Promise<{ key: Buffer; value: T }[]>;
```

Comment:
These really don't look like iterators. We should rather rename to `toArray` and `toArrayWithSchema`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Iterators_and_Generators
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols#the_iterable_protocol